packages/bpk-component-navigation-bar/src/BpkNavigationBar.js

```diff
@@ -52,7 +52,10 @@ const BpkNavigationBar = (props: Props) => {
     ...rest
   } = props;
 
-  const titleId = `${id}-bpk-navigation-bar-title`;
+  // If the title is a component that sets its own id we want the aria-labelledby on the nav to match this so it can find the element
+  // Otherwise if its just a string we set the id on the title component.
+  const titleId =
+    typeof title === 'string' ? `${id}-bpk-navigation-bar-title` : id;
```

Comment:
I'm not entirely clear on the reason for this type check?

Reply:
So if the the title passed is not a string and a component 

E.g. H2 or BpkText then we would already be passing an id which then when passed from something like the modal which has an `aria-labelledby` if we didn't use the id that was passed it would have the extra string appended to the id which would then mismatch with the `aria-labelledby` in the parent component so throws an a11y error.

Maybe that isn't too clear in the comment above, happy to have a call to discuss/clarify